R/zchunk_L2242.land_input_4_irr_mgmt.R

```diff
@@ -1,90 +1,119 @@
     return(c("L2242.LN4_Logit",
              "L2242.LN4_NodeGhostShare",
              "L2242.LN4_NodeIsGhostShareRel"))
   } else if(command == driver.MAKE) {
 
+    # silence package check notes
+   GLU_name <- LandLeaf <- LandNode4 <- NULL
+
     all_data <- list(...)[[1]]
 
     # Load required inputs
-    L2241.LN4_Logit <- get_data(all_data, "L2241.LN4_Logit")
-    L2241.LN4_NodeGhostShare <- get_data(all_data, "L2241.LN4_NodeGhostShare")
-    L2241.LN4_NodeIsGhostShareRel <- get_data(all_data, "L2241.LN4_NodeIsGhostShareRel")
+    A_LandNode_logit_irr <- get_data(all_data, "aglu/A_LandNode_logit_irr")
+    L223.LN3_MgdAllocation_bio <- get_data(all_data, "temp-data-inject/L223.LN3_MgdAllocation_bio")
+    L223.LN3_MgdAllocation_crop <- get_data(all_data, "temp-data-inject/L223.LN3_MgdAllocation_crop")
+    L223.LN3_LeafGhostShare <- get_data(all_data, "temp-data-inject/L223.LN3_LeafGhostShare")
+    L223.LN3_LeafIsGhostShareRel <- get_data(all_data, "temp-data-inject/L223.LN3_LeafIsGhostShareRel")
+
+    # L2242.LN4_Logit: Logit exponent of the fourth land nest by region
+    # There are no technologies that are disaggregated to irrigated and rainfed but not to lo- and hi-input techs,
+    # so here we only write out the logit exponent for the irrigated/rainfed node competition.
+    # Use the cropland and bioenergy allocation tables to establish which region/GLU/node combinations are available
+    L223.LN3_MgdAllocation_crop %>%
+      select(one_of(LEVEL2_DATA_NAMES[["LN3_Leaf"]])) %>%
+      unique %>%
+      bind_rows(unique(select(L223.LN3_MgdAllocation_bio, one_of(LEVEL2_DATA_NAMES[["LN3_Leaf"]])))) %>%
+      # What was a leaf for level3 is now a node, as it will have the 4th level nested under it
+      rename(LandNode4 = LandLeaf) %>%
+      # Modify land node variable, prepare to separate the GLU name
+      mutate(LandNode4 = sub("Root_Tuber", "RootTuber", LandNode4),
+             LandNode4 = sub("biomass_tree", "biomasstree", LandNode4),
+             LandNode4 = sub("biomass_grass", "biomassgrass", LandNode4)) %>%
+      separate(LandNode4, c("LandNode4", "GLU_name")) %>%
+      mutate(logit.year.fillout = min(BASE_YEARS),
+             # Modify land node variable to match in logit exponent values
+             LandNode4 = sub("RootTuber", "Root_Tuber", LandNode4),
+             LandNode4 = sub("biomasstree", "biomass_tree", LandNode4),
+             LandNode4 = sub("biomassgrass", "biomass_grass", LandNode4)) %>%
+      # Match in logit exponent values, use left_join instead because the logit.type variable are NAs, drop later
+      left_join(A_LandNode_logit_irr, by = c("LandNode4" = "LandNode")) %>%
+      mutate(LandNode4 = paste(LandNode4, GLU_name, sep = "_")) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["LN4_Logit"]])) %>%
+      # Note: The na.omit is intended to get rid of a tailing node-rename table
+      # that may be appended to level2 land alloator tables
+      na.omit ->
```

Comment:
By adding the `logit.type` which are NAs, `na.omit` drops every single row in the tibble. 
It currently does not make any difference. Is it safe to drop this line? @bpbond @pralitp

Reply:
Judging by that comment and that we no longer to worry about any `node-rename` issues in DSR I would say yes go ahead and drop it and the comment which is no longer applicable.